daemon/containerd/image_manifest.go

```diff
@@ -149,3 +218,30 @@ func readManifest(ctx context.Context, store content.Provider, desc ocispec.Desc
 
 	return mfst, nil
 }
+
+// ImagePlatform returns the platform of the image manifest.
+// If the manifest descriptor entry in the index has a nil platform, it will be read from the config.
+func (m *ImageManifest) ImagePlatform(ctx context.Context) (ocispec.Platform, error) {
+	if m.cachedImagePlatform != nil {
+		return *m.cachedImagePlatform, nil
+	}
+
+	target := m.Target()
+	if target.Platform != nil {
+		return *target.Platform, nil
+	}
+
+	configDesc, err := m.Config(ctx)
+	if err != nil {
+		return ocispec.Platform{}, err
+	}
+
+	var out ocispec.Platform
+	err = readConfig(ctx, m.ContentStore(), configDesc, &out)
+	if err != nil {
+		return ocispec.Platform{}, err
+	}
+
+	m.cachedImagePlatform = &out
```

Comment:
We don't need a mutex on this, right? :eyes: (I'm guessing there's one further out or that these `ImageManifest` objects aren't shared, but the latter would mean this cache is never used so keeping my fingers crossed for one further out, but that also means it's maybe not ideal that this is a potentially heavy method, right?)

Reply:
In general the `ImageManifest` objects are short-lived and are not persisted anywhere, so there shouldn't be a case where they are accessed concurrently.

But actually, you're right that the caching isn't actually used _any more_ (at some point the `ImagePlatform` was called multiple times in the same function in different places). So might as well just drop that, especially since the branch where we reach out to the image config is rather unexpected (as the target is the descriptor from the image index which SHOULD have a platform set).

Reply:
Removed